src/include/miopen/performance_config.hpp

```diff
@@ -68,6 +68,20 @@ struct PerfConfigBase : PerfConfig
     }
 };
 
+template <class Derived>
+struct PerfConfigBaseCK : PerfConfig
+{
+    void Serialize(std::ostream& stream) const final
+    {
+        serialize::SerDes<'\0'>::Serialize(static_cast<const Derived&>(*this), stream);
+    }
```

Comment:
The first solutions makes the most sense to me, didn't realize that just eliminating the abstraction for deserialization/serialization would lead to such a simple solution. Thanks.